OpenRA.Game/Map/MapCache.cs

```diff
@@ -176,34 +175,31 @@ void LoadAsyncInternal()
 						bitmap = Minimap.RenderMapPreview(modData.DefaultRules.TileSets[p.Map.Tileset], p.Map, modData.DefaultRules, true);
 					}
 
-					// Note: this is not generally thread-safe, but it works here because:
-					//   (a) This worker is the only thread writing to this sheet
-					//   (b) The main thread is the only thread reading this sheet
-					//   (c) The sheet is marked dirty after the write is completed,
-					//       which causes the main thread to copy this to the texture during
-					//       the next render cycle.
-					//   (d) Any partially written bytes from the next minimap is in an
-					//       unallocated area, and will be committed in the next cycle.
-					try
+					Game.RunAfterTick(() =>
 					{
-						p.SetMinimap(sheetBuilder.Add(bitmap));
-					}
-					finally
-					{
-						if (createdPreview)
-							bitmap.Dispose();
-					}
+						try
+						{
+							p.SetMinimap(sheetBuilder.Add(bitmap));
```

Comment:
Check the stacktrace in #8049. It tries creating the managed buffer from the worker thread and dies.

This will be as a result of the memory saving measures from #6756. Since we no longer keep the managed buffer around forever, the old comment hasn't been correct since then.